src/query/algorithms/entityresolution/EntityResolver.cpp

```diff
@@ -0,0 +1,502 @@
+//    string partitionID = trainargs[2];
+//
+//    //Read attributes from file
+//    std::ifstream dataFile;
+//    cout << "reading file" << endl;
+//
+//    filePath = utils.getJasmineGraphProperty("org.jasminegraph.entity.location");
+//    dataFile.open("/home/damitha/ubuntu/entity_data/entityData.txt", std::ios::binary | std::ios::in);
+//
+//    char splitter;
+//    string line;
+//
+//    std::getline(dataFile, line);
+//    splitter = ' ';
+//
+//    cout << "Getting data" << endl;
+//    while (!line.empty()) {
+//
+//        int strpos = line.find(splitter);
+//        string nodeID = line.substr(0, strpos);
+//        string weakIDStr = line.substr(strpos + 1, -1);
+//        entityData[stoi(nodeID)] = utils.split(weakIDStr, splitter);
+//        cout << line << endl;
+//
+//
+//        std::getline(dataFile, line);
+//        while (!line.empty() && line.find_first_not_of(splitter) == std::string::npos) {
+//            std::getline(dataFile, line);
+//        }
+//    }
+//    dataFile.close();
+//
+//    //Read edgelist from file
+//    std::ifstream edgeFile;
+//    std::cout << "reading file" << endl;
+//
+//    edgefilePath = utils.getJasmineGraphProperty("org.jasminegraph.server.instance.trainedmodelfolder")
+//                   + graphID + "_" + partitionID;
+//    edgeFile.open(edgefilePath);
+//    std::getline(edgeFile, line);
+//
+//    splitter = ' ';
+//
+//    cout << "Getting neighbourhood data" << endl;
+//    while (!line.empty()) {
+//
+//        int strpos = line.find(splitter);
+//        string vertex1 = line.substr(0, strpos);
+//        string vertex2 = line.substr(strpos + 1, -1);
+//        if (neighborhoodData.size() <= 5) {
+//            neighborhoodData[stoi(vertex1)].emplace_back(stoi(vertex2));
+//        }
+//
+//
+//        std::getline(edgeFile, line);
+//        while (!line.empty() && line.find_first_not_of(splitter) == std::string::npos) {
+//            std::getline(edgeFile, line);
+//        }
+//    }
+//    edgeFile.close();
+//
+//    map<int, string> attrFilters = createFilters(entityData, 256, 4);
+//
+//    attrFilterPath =
+//            utils.getJasmineGraphProperty("org.jasminegraph.entity.location") + "attrfilters_" + graphID + "_" +
+//            partitionID + ".txt";
+//    structFilterPath =
+//            utils.getJasmineGraphProperty("org.jasminegraph.entity.location") + "structfilters_" + graphID + "_" +
+//            partitionID + ".txt";
+//
+//    //Write attr and struct filters separately into files
+//    writeToFile(attrFilterPath, attrFilters);
+////    writeBloomFiltersToFile(structFilterPath, structFilters);
+//
+//}
+
+Mat<short> EntityResolver::generateCRVs(int minhashSize, int noClusters) { //minHash def 100
+    Mat<short> CRVs(minhashSize, noClusters);
+    for (int i = 0; i < noClusters; i++) {
+        //Load cluster file into memory
+        string filename = "/root/CLionProjects/EntityResolution/cluster" + to_string(i) + "filters.txt";
```

Comment:
Need to store these files in JasmineGraph worker's local data directory

Reply:
Location updated

Reply:
filename changed to localstore folder location

Reply:
Ok